producer-kafka/README.adoc

```diff
@@ -0,0 +1,407 @@
+= Using the Kafka CDC Producer
+
+This extension provides support for sending Change Data Capture(CDC) events to an Apache Kafka®.
+It is designed to work with the Stargate.
+
+== Kafka CDC - Stargate integration
+
+This extension is designed to integrate with Stargate. It uses an link:https://www.osgi.org/[OSGi] framework for registering the extension within Stargate.
+The `KafkaProducerActivator` class handles the registration process. When you put the `producer-kafka-{version}.jar` in the
+`/stargate-lib` directory it will be automatically picked, and the activator logic is invoked.
+
+This extension has one service required: `Metrics` - this is provided by the `CoreActivator` in the link:../core[core] module.
+
+== Architecture
+
+The main entry point for this CDC connector is a link:../producer-kafka/src/main/java/io/stargate/producer/kafka/KafkaCDCProducer.java[KafkaCDCProducer].
+It implements the SchemaAwareCDCProducer interface that serves as an integration layer between Stargate's persistence and CDC extension.
+
+1. `init()` - takes the map of configuration settings that are passed from Stargate to the CDC connector. The configuration settings are described in <<Configuration>>
```

Comment:
I think that from the user pov, a high level overview is more beneficial than a method by method rundown (maybe this kind of doc belongs in code).

Reply:
For the CDC-extension, user pov is a stargate. The stargate is the main user of this extension. It cannot be used outside of the stargate. So this method rundown shows the contract between this producer and stargate. If we have an external contributor to this particular Kafka-CDC-extension, it will be beneficial to contributors to have that information.